core/src/main/java/com/sequenceiq/cloudbreak/service/publicendpoint/FreeIPAEndpointManagementService.java

```diff
@@ -0,0 +1,117 @@
+                String domain = getDomainFromStack(stack);
+
+                if (StringUtils.isNotEmpty(loadBalancer.getDns())) {
+                    AddDnsCnameRecordRequest request = new AddDnsCnameRecordRequest();
+                    request.setCname(endpoint);
+                    request.setDnsZone(domain);
+                    request.setTargetFqdn(StringUtils.appendIfMissing(loadBalancer.getDns(), DOMAIN_PART_DELIMITER));
+                    request.setEnvironmentCrn(stack.getEnvironmentCrn());
+                    dnsV1Endpoint.addDnsCnameRecord(request);
+                } else if (StringUtils.isNotEmpty(loadBalancer.getIp())) {
+                    AddDnsARecordRequest request = new AddDnsARecordRequest();
+                    request.setHostname(endpoint);
+                    request.setDnsZone(domain);
+                    request.setIp(loadBalancer.getIp());
+                    request.setEnvironmentCrn(stack.getEnvironmentCrn());
+                    dnsV1Endpoint.addDnsARecord(request);
+                } else {
+                    LOGGER.error("Unable to find DNS or IP for load balancer. Load balancer will not be registered with FreeIPA.");
+                }
+            }
+        }
+    }
+
+    public void deleteLoadBalancerDomainFromFreeIPA(Stack stack) {
+        if (manageCertificateAndDnsInPem()) {
+            Set<LoadBalancer> loadBalancers = loadBalancerPersistenceService.findByStackId(stack.getId());
+            if (!loadBalancers.isEmpty()) {
+                Optional<LoadBalancer> loadBalancerOptional = selectLoadBalancer(loadBalancers);
+                if (loadBalancerOptional.isEmpty()) {
+                    LOGGER.debug("Unabled to determine load balancer type. Load balancer will not be deleted from FreeIPA.");
+                    return;
+                }
+
+                LoadBalancer loadBalancer = loadBalancerOptional.get();
+                String endpoint = loadBalancer.getEndpoint();
+                String domain = getDomainFromStack(stack);
+
+                if (StringUtils.isNotEmpty(loadBalancer.getDns())) {
+                    dnsV1Endpoint.deleteDnsCnameRecord(stack.getEnvironmentCrn(), domain, endpoint);
+                } else if (StringUtils.isNotEmpty(loadBalancer.getIp())) {
+                    dnsV1Endpoint.deleteDnsARecord(stack.getEnvironmentCrn(), domain, endpoint);
+                } else {
+                    LOGGER.debug("Unable to find DNS or IP for load balancer. Load balancer will not be deleted from FreeIPA.");
+                }
+            }
+        }
+    }
+
+    private String getDomainFromStack(Stack stack) {
+        String userCrn = ThreadBasedUserCrnProvider.getUserCrn();
+        DetailedEnvironmentResponse environment = environmentClientService.getByCrn(stack.getEnvironmentCrn());
+        return getDomainNameProvider().getDomainName(environment.getName(), getWorkloadSubdomain(userCrn));
+    }
```

Comment:
I'm not sure if you really need this. If this domain is the same which is handled by FreeIPA you could leave that field blank and FreeIPA will take care of it.

Reply:
I'm also not sure I need it either. But I don't think it hurts to have it and I'd rather err on the side of making sure it's defined explicitly instead of hoping it's the same domain as in FreeIPA.

Reply:
then I think you should invest more time into this: what kind of domain could be returned by this code, can it be different than the domain in freeipa? can it be different than `Stack` entity domain field? If we can't tell why we need this and how it should work, I won't expect us to understand it a few month later.

Reply:
I talked to Jamison about how the DNS zones work. I've removed this method and am not passing the dnsZone in the requests anymore. I ran a test and everything is still being registered correctly without explicitly setting the dnsZone.